### PR TITLE
mimxrt: Stop a previous timer on channel reuse.

### PR DESCRIPTION
**Pull request by robert-hh**
As suggested by Damien. Doing it now, before it fades away in the past.

### CODE REVIEW

File: ports/mimxrt/machine_timer.c [hunk 1]
```diff
@@ -148,6 +148,12 @@ STATIC mp_obj_t machine_timer_make_new(const mp_obj_type_t *type, size_t n_args,
         mp_raise_ValueError(MP_ERROR_TEXT("Timer does not exist"));
     }
 
+    // check, if a timer exists at that channel and stop it first
+    if (MP_STATE_PORT(timer_table)[id] != NULL) {
+        PIT_StopTimer(PIT, channel_no[id]);
+        MP_STATE_PORT(timer_table)[id] = NULL;
+    }
```
Comment (dpgeorge) [inline ports/mimxrt/machine_timer.c:155]: I'd suggest to just re-use the object:
```c
if (MP_STATE_PORT(timer_table)[id] == NULL) {
    self = m_new_obj_with_finaliser(machine_timer_obj_t);
    MP_STATE_PORT(timer_table)[id] = self;
} else {
    self = MP_STATE_PORT(timer_table)[id];
}
```
That way the hardware timer is mapped 1-1 to an object, and you can retrieve the existing object via `machine.Timer(id)`.  This is how other ports work.

**Comment by dpgeorge** [comment]
This looks better now, but I've yet to test it.

Do you also need to remove the WFE from the other locations, or is it enough to remove it from the EVENT_POLL_HOOK only?

**Comment by robert-hh** [comment]
The only place where I ever changed it was at the EVENT_POLL_HOOK.

**Comment by dpgeorge** [comment]
Merged in 5226d6e1ee7282bf40a45708b98067446b157bce, cdd95ce73705ec8dc5249a7e22092a8ce374576d and d79105d7c0f68e65ff175b752d9895241e355c08 (with a fix so that init can be called again after deinit).

**Pull request closed without merging**